great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -3,18 +3,20 @@
 import json
 import logging
 import os
-from abc import ABC, abstractmethod
-from typing import Mapping, Optional, Union
+from abc import abstractmethod
+from typing import Dict, Mapping, Optional, Union
 
+from great_expectations.core.config_peer import ConfigPeer
 from great_expectations.data_context.types.base import (
     DataContextConfig,
     anonymizedUsageStatisticsSchema,
 )
+from great_expectations.data_context.util import substitute_all_config_variables
 
 logger = logging.getLogger(__name__)
 
 
-class AbstractDataContext(ABC):
+class AbstractDataContext(ConfigPeer):
```

Comment:
`ConfigPeer` is a `ABC` that ensures that objects inheriting from it can be instantiated using `instantiate_class_from_config()` and supports various modes of serialization (YAML, JSON, etc). 

Currently `BaseDataContext` inherits from it, and there is a future `TODO` item to remove the dependency as part of decoupling serialization from configuration. 

This PR proposes bumping the `ConfigPeer` dependency up to `AbstractDataContext` until it can be removed as part of future work.